hoomd/hpmc/pytest/test_muvt.py

```diff
@@ -159,3 +159,144 @@ def test_insertion_removal(device, simulation_factory,
 
     # make a wild guess: there be B particles
     assert (muvt.N['B'] > 0)
+
+@pytest.mark.skipif(not hoomd.version.llvm_enabled, reason='LLVM not enabled')
+def test_jit_remove_insert(device, simulation_factory,
+                           one_particle_snapshot_factory):
+    """Test that MuVT considers a cpp potential when removing particles by adding a high energy particles that should always be removed by the updater while inserting particles to the top of the box."""
+    sim = simulation_factory(
+        one_particle_snapshot_factory(particle_types=['A'],
+                                 dimensions=3,
+                                 position=(0, 0, -5),
+                                 orientation=(1, 0, 0, 0),
+                                 L=20)
+                                    )
+
+    sphere_radius = 0.6 
+    mc = hoomd.hpmc.integrate.Sphere(default_d=0.0, default_a=0.0)
+    mc.shape['A'] = dict(diameter=2*sphere_radius)
+
+    # code returns infinity if the particle center goes past the center of the box
+
+    center_wall = '''
+        if (r_i.z < 0. || r_i.z > box.getL().z/2)
+            return INFINITY;
+        else
+            return 0.0f;
+    '''
+    cpp_external_potential = hoomd.hpmc.external.user.CPPExternalPotential(
+        code=center_wall)
+
+    mc.external_potential = cpp_external_potential
+
+    sim.operations.integrator = mc
+
+    muvt = hoomd.hpmc.update.MuVT(trigger=hoomd.trigger.Periodic(1),
+                                  transfer_types=['A'])
+    muvt.fugacity['A'] = 1e6 
+    sim.operations.updaters.append(muvt)
+    sim.run(1000)
+    snapshot = sim.state.get_snapshot()
+    pos = snapshot.particles.position
+    
+    # We should have inserted particles successfully
+    assert muvt.insert_moves[0] > 0
+
+    # We should have successfully removed the high energy particle
+    assert muvt.remove_moves[0] > 0
+
+    # We should have successfully inserted only to the top of the box and removed the high energy particle
+    assert  numpy.min(pos[:,2]) >= 0. 
+
+    # We should have added more than one particle to the box
+    assert  len(pos) > 1 
+
+def test_plane_wall_insertion(device, simulation_factory,
```

Comment:
I like this test, however I would like to see this test and the following two combined into a single test which is parametrized on the type of the wall.

Reply:
Chatted with @joaander about this and we think we should keep the tests this way for ease of readability, time writing the test, and simplicity.